src/components/modals/CreateAntModal/CreateAntModal.tsx

```diff
@@ -0,0 +1,594 @@
+    if (walletAddress) {
+      if (!ant.owner) {
+        ant.owner = walletAddress.toString();
+      }
+      if (!ant.controller) {
+        ant.controller = walletAddress.toString();
+      }
+      setDetails();
+    }
+  }, [walletAddress, ant, workflowStage]);
+
+  function setDetails() {
+    //  eslint-disable-next-line
+
+    const consolidatedDetails: any = {
+      name: ant.name,
+      ticker: ant.ticker,
+      targetID: ant.records['@'].transactionId,
+      ttlSeconds: ant.records['@'].ttlSeconds?.toString(),
+      controller: ant.controller,
+      owner: ant.owner,
+    };
+
+    const rows = Object.keys(consolidatedDetails).reduce(
+      (details: any, attribute: string, index: number) => {
+        const detail = {
+          attribute,
+          value: consolidatedDetails[attribute],
+          editable: EDITABLE_FIELDS.includes(attribute),
+          action: ACTIONABLE_FIELDS[attribute],
+          key: index,
+        };
+        details.push(detail);
+        return details;
+      },
+      [],
+    );
+    setRows(rows);
+  }
+
+  function handleStateChange() {
+    try {
+      if (!modifiedValue) {
+        throw new Error('Value is undefined');
+      }
+
+      if (editingField == 'name') {
+        ant.name = modifiedValue.toString();
+      }
+      if (editingField == 'ticker') {
+        ant.ticker = modifiedValue.toString();
+      }
+      if (editingField == 'owner') {
+        ant.owner = modifiedValue.toString();
+      }
+      if (editingField == 'controller') {
+        ant.controller = modifiedValue.toString();
+      }
+      if (editingField == 'targetID') {
+        ant.records = {
+          '@': {
+            transactionId: modifiedValue.toString(),
+          },
+        };
+      }
+      if (editingField == 'ttlSeconds' && typeof modifiedValue === 'number') {
+        ant.records = {
+          '@': {
+            ttlSeconds: modifiedValue,
+          },
+        };
+      }
+      setEditingField(undefined);
+      setModifiedValue(undefined);
+    } catch (error) {
+      console.error(error);
+      setEditingField(undefined);
+      setModifiedValue(undefined);
+    }
+  }
+
+  async function deployAnt(state: ANTContractJSON) {
+    try {
+      setIsPostingTransaction(true);
+      if (!state) {
+        throw new Error('No state provided, cannot deploy ANT contract');
+      }
+      // perform checks, try/catch
+      const pendingTXId = await arweaveDataProvider.deployContract({
+        srcCodeTransactionId: new ArweaveTransactionID(
+          DEFAULT_ANT_SOURCE_CODE_TX,
+        ),
+        initialState: state,
+      });
+      if (!pendingTXId) {
+        throw new Error('Failed to deploy ANT contract');
+      }
+      setAntContractId(new ArweaveTransactionID(pendingTXId));
+      setIsPostingTransaction(false);
+      console.log(pendingTXId);
+
+      return pendingTXId;
+    } catch (error) {
+      console.error(error);
+      setIsPostingTransaction(false);
+    }
```

Comment:
use `finally` to `setIsPositingTransaction`

Reply:
added